### ISSUE TITLE
[BUG] Using Realm with external storage

### ISSUE THREAD

**Issue by mnlbox** [realm/realm-js#2062]
## Goals

I want to use sdcard on Android to save and load Realm data.
## Expected Results

The file can be created and loaded without problems using app's folder on external storage.
## Actual Results

Unfortunately, when running Realm.open, it throws an exception of "Read-only file system", even after giving runtime permission to read and write external storage. I can see the file was created at the specified path.

![screenshot_1524455457](https://user-images.githubusercontent.com/2258901/39106498-8833f4c8-4692-11e8-8fe0-4c94f8b3ae23.png)
## Steps to Reproduce

It's an old issue that still not fixed. Please also see: #1760 
Use below code sample

## Code Sample
I open my database like this in react-native:
```
    componentWillMount() {
        Realm.open({
            schema: [ Constant.BookSchema ],
            path: '/storage/sdcard0/BookDB/db.realm'
        }).then(realm => {
            // some code here
        });
    }
```
I also add this two line to my `AndroidManifest.xml`:
```
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
```

It's create below files in this location `/storage/sdcard0/BookDB`:
```
db.realm.managment
    access_control.control.mx
    access_control.write.mx
db.realm
db.realm.lock
```
But It's return red error page with `Read-only file system` error. (something like above image)
My device just have internal storage with `/storage/sdcard0` location and it's not have any external SD card. (with Android 5.1.1 OS)

## Version of Realm and Tooling

    * Realm JS SDK Version: 2.15.3

    * Node or React Native: 0.55.4

    * Client OS & Version: Android API 25, 27

    * Which debugger for React Native: None

**Comment by mnlbox** [comment]
@ianzitow @kneth, @bmunkholm

**Comment by mnlbox** [comment]
Anyone here?
Why no body use `path` parameter?

**Comment by kneth** [comment]
@mnlbox From API 23 you need to ask the user for granting permissions: https://facebook.github.io/react-native/docs/permissionsandroid

**Comment by mnlbox** [comment]
@kneth So why It's create below files in this location `/storage/sdcard0/BookDB`: :thinking: :thinking: 
```
db.realm.managment
    access_control.control.mx
    access_control.write.mx
db.realm
db.realm.lock
```

**Comment by mnlbox** [comment]
@kneth Any idea?

**Comment by mnlbox** [comment]
Anyone here?

**Comment by kneth** [comment]
@mnlbox No real good suggestions. Can you create a minimal app which reproduce the issue? It can help us understand what the cause is.

**Comment by Finneah** [comment]
Same problem here. Any fixes?

**Comment by mnlbox** [comment]
@kneth I try to provide some minimal reproduce repo for test soon.

**Comment by Finneah** [comment]
@kneth here is a small example: https://github.com/Finneah/testRealmOnReactNative

**Comment by cmelchior** [comment]
A few observations:

- Removing Realm completely and just writing something random to the `Download` folder works fine. Hinting at Realm trying to write something where it shouldn't.
- I tried to create a pure Java app with the same behavior of requesting permissions and then creating a Realm in the `Download` folder. This works fine, which indicates a problem more directly with RealmJS or ReactNative, but since Java and JS instantiate things slightly differently, I'm still investigating that.

**Comment by mnlbox** [comment]
@cmelchior Thanks for your work. Any news?

**Comment by mnlbox** [comment]
Guys I'm still waiting for a fix after 6 month 😞
Any plan to fix this?

@cmelchior

**Comment by cmelchior** [comment]
@mnlbox I still haven't found the root cause for this. 

However, I noticed that opening the Realm fails to create a number of files:
```
// Locally
./test.realm
./test.realm.lock
./test.realm.note
./test.realm.management/access_control.control.mx
./test.realm.management/access_control.new_commit.cv
./test.realm.management/access_control.pick_writer.cv
./test.realm.management/access_control.write.mx

// On device
./test.realm
./test.realm.lock
./test.realm.management/access_control.control.mx
./test.realm.management/access_control.write.mx

// On device missing files
./test.realm.note
./test.realm.management/access_control.new_commit.cv
./test.realm.management/access_control.pick_writer.cv
```

I guess the reason you see the `[Error: Read-only file system]` error is because of the missing files. But I haven't been able to determine yet why.

Not, it appears not to be possible to create files anywhere on the SDCard. I also tried the app external data dir: `/sdcard/Android/data/com.test/files` which also fails.

**Comment by cmelchior** [comment]
Fixed in #2235 
A release will be made later today

**Comment by duccyf** [comment]
Thanks @cmelchior.
